assets/js/phoenix_live_view.js

```diff
@@ -968,6 +971,12 @@ export let DOM = {
 
   isFormInput(el){ return /^(?:input|select|textarea|button)$/i.test(el.tagName) },
 
+  syncAttrsToProps(el){
+    if(el instanceof HTMLInputElement && CHECKABLE_INPUTS.indexOf(el.type.toLocaleLowerCase()) >= 0){
+      el.checked = el.getAttribute("checked") !== null
+    }
+  },
```

Comment:
As noted for future readers, this would require `<input type="checkbox" <%= @checked && "checked" %> />` and not work for `<input type="checkbox" />`.